src/mam4xx/modal_aer_opt.hpp

```diff
@@ -0,0 +1,1084 @@
+  // FIXME
+  refr = crefin.real();
+  refi = crefin.imag();
+
+} // calc_refin_complex
+
+KOKKOS_INLINE_FUNCTION
+void compute_factors(const int prefri, const Real ref_ind,
+                     const Real *ref_table, int &ix, Real &tt) {
+  // Compute factors for the real or imaginary parts
+
+  // prefri, ncol
+  // ref_table(:) !refractive index table [unitless]
+  // ref_ind(:)   !refractive index       [unitless]
+
+  // FORTRAN refactor note: "do_print" is kept to maintain code consistenty with
+  // the original code. THis can be removed from the C++ ported code
+  //  logical, intent(in), optional :: do_print ! to print log msg or not
+
+  // !intent-inouts/outs
+  // ix(:)
+  // tt(:)
+  // BAD CONSTANT
+  constexpr Real threshold = 1.e-20;
+  constexpr Real zero = 0;
+  ix = 1;
+  tt = zero;
+  if (prefri > 1) {
+
+    int ii = 0;
+    for (ii = 0; ii < prefri; ++ii) {
+      if (ref_ind < ref_table[ii]) {
+        break;
+      }
+    } // ii
+    // FIXME: check Fortran to C++ indexing conversion
+    ix = haero::max(ii - 1, 0);
+    const int ip1 = haero::min(ix + 1, prefri - 1);
+    const Real dx = ref_table[ip1] - ref_table[ix];
+    if (haero::abs(dx) > threshold) {
+      tt = (ref_ind - ref_table[ix]) / dx;
+      // FIXME: I did not port the following:
+      // if (present(do_print) .and. do_print) then
+      //       if(tt(ic) < 0._r8 .or. tt(ic) > 1._r8) then
+      //          write(iulog,*)
+      //          'tt,ref_ind,ix,ref_table,dx=',tt(ic),ref_ind(ic),ix(ic),ref_table(ix(ic)),dx
+      //       endif
+      //    endif
+
+    } // (abs(dx) > threshold)
+
+  } // if(prefri > 1)
+
+} // compute_factors
+
+KOKKOS_INLINE_FUNCTION
+void binterp(const View3D &table, const Real ref_real, const Real ref_img,
+             const Real ref_real_tab[prefr], const Real ref_img_tab[prefi],
+             int &itab, int &jtab, Real &ttab, Real &utab, Real coef[ncoef],
+             const int itab_1) {
+  /*------------------------------------------------------------------------------
+   Bilinear interpolation along the refractive index dimensions
+   of the table to estimate Chebyshev coefficients at an
+   intermediate refractive index.
+
+   In short wave, the first call computes itab, jtab, ttab, utab and coef.
+   The subsequent calls use itab, jtab, ttab and utab as inputs and compute coef
+
+   In long wave, we have just one call to compute itab,jtab,ttab, utab and coef
+  ------------------------------------------------------------------------------*/
+
+  // intent-ins
+  // ncol
+  // table(ncoef,prefr,prefi)
+  // ref_real(pcols), ref_img(pcols) !real and imganinary parts of refractive
+  // indices [unitless] ref_real_tab(prefr), ref_img_tab(prefi) !real and
+  // imganinary table refractive indices [unitless]
+
+  // intent-inouts/outs
+  // itab(pcols), jtab(pcols)
+  // ttab(pcols), utab(pcols)
+  // coef(pcols,ncoef) !coefficient interpolated bilinearly
+  // FIXME; maybe we need to loop over cols
```

Comment:
I hope not; that would break the single-column paradigm we've been using!

Reply:
Old comment. I deleted it.